app/src/main/java/atlant/moviesapp/fragments/MovieFavoritesFragment.java

```diff
@@ -0,0 +1,235 @@
+
+                    View itemView = viewHolder.itemView;
+                    float height = (float) itemView.getBottom() - (float) itemView.getTop();
+                    float width = height / 3;
+                    if(dX > 0){
+                        p.setColor(getResources().getColor(R.color.standardYellow));
+                        t.setColor(Color.parseColor("#FFFFFF"));
+                        RectF background = new RectF((float) itemView.getLeft(), (float) itemView.getTop(), dX,(float) itemView.getBottom());
+                        c.drawRect(background,p);
+                        RectF icon_dest = new RectF((float) itemView.getLeft() + width ,(float) itemView.getTop() + width,(float) itemView.getLeft()+ 2*width,(float)itemView.getBottom() - width);
+                        c.drawText("Remove",(float) itemView.getLeft() + width,(float) itemView.getTop() + width,t);
+                    } else {
+                        p.setColor(getResources().getColor(R.color.standardYellow));
+                        t.setColor(Color.parseColor("#FFFFFF"));
+                        RectF background = new RectF((float) itemView.getRight() + dX, (float) itemView.getTop(),(float) itemView.getRight(), (float) itemView.getBottom());
+                        c.drawRect(background,p);
+                        RectF icon_dest = new RectF((float) itemView.getRight() - 2*width ,(float) itemView.getTop() + width,(float) itemView.getRight() - width,(float)itemView.getBottom() - width);
+                        c.drawText("Remove",(float) itemView.getLeft() + width,(float) itemView.getTop() + width,t);
+
+                    }
+                }
+                super.onChildDraw(c, recyclerView, viewHolder, dX, dY, actionState, isCurrentlyActive);
+            }
+        };
+
+        ItemTouchHelper itemTouchHelper = new ItemTouchHelper(simpleItemTouchCallback);
+        itemTouchHelper.attachToRecyclerView(recyclerView);
+
+        return v;
+    }
+
+    @Override
+    public void showError() {
+        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
+        builder.setMessage("Error loading data, please check your connection")
+                .setTitle("Error");
+
+        builder.setPositiveButton("OK", new DialogInterface.OnClickListener() {
+            public void onClick(DialogInterface dialog, int id) {
+                // User clicked OK button
+            }
+        });
+        AlertDialog dialog = builder.create();
+        dialog.show();
+
+    }
+
+    @Override
+    public void showMovies(List<Movie> data) {
+        if (!(data.size() > 0)) {
+            adapter.setMoreDataAvailable(false);
+        } else {
+            favoriteMovies.addAll(data);
+            adapter.notifyDataChanged();
+        }
+
+    }
+
+    @Override
+    public void showTvShows(List<TvShow> data) {
+
+    }
+
+    @Override
+    public void showProgress() {
+        progressBar.setVisibility(View.VISIBLE);
```

Comment:
Its good practice to check if the progress is not null and if it isn't already in the VISIBLE state